src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp

```diff
@@ -1367,4 +1367,41 @@ namespace OpenMS
     }
   }
 
+  void OPXLHelper::filterCandidatesByTags(std::vector <OPXLDataStructs::ProteinProteinCrossLink>& candidates, std::set<std::string>& tags)
+  {
+    std::vector <OPXLDataStructs::ProteinProteinCrossLink> filtered_candidates;
+    std::vector<bool> selection(candidates.size(), false);
+    
+    // brute force string comparisons for now
+    for (Size i = 0; i < candidates.size(); ++i)
+    {
+      bool candidate_added(false);
+      OPXLDataStructs::ProteinProteinCrossLink candidate = candidates[i];
+      String alpha = candidate.alpha->toUnmodifiedString();
+      String beta;
+      if (candidate.beta)
+      {
+        beta = candidate.beta->toUnmodifiedString();
+      }
+      for (std::string tag : tags)
```

Comment:
you are just searching unmodified tags against unmodified proteins/peptides, right?
Could be faster to build a search-trie for the tags and push the PPs across them and stop at the first hit -- should be very fast. Building the trie is also very fast.